docs/getting-started/01-overview/03.dependency-injection.md

```diff
@@ -126,33 +138,64 @@ class ConsoleLogHandler implements LogHandler {
 
 @injectable()
 class Logger {
-    constructor(@injectAll("LogHandler") private readonly logHandlers: LogHandlerInterface[]) {} // <-- It would be nice to be able to have all the classes that implement LogHandler 
+    constructor(@injectAll("LogHandler") private readonly logHandlers: LogHandlerInterface[]) {} 
 }
 
 ```
 
-**Side note, due to how TSyringe works, if you use `@injectAll` and there are no classes, TSryinge will throw. Therefore, if you use `@injectAll`, makes sure at least one of your class has the `@tag` decorator.**
+**Side note, due to how TSyringe works, if you use `@injectAll` and there are no classes, TSryinge will throw an error.
+Therefore, if you use `@injectAll`, makes sure at least one of your class has the `@tag` decorator.**
 
 **Tags coupled with interfaces**
 
-// todo write here how you can use tags to inject Interfaces.
+Similarly to above, the `@tag` can be used with Interfaces. Injecting a class that implements an Interface is a really
+useful pattern.
+
+```
+interface LoggerInterface {
+    log();
+}
+
+@tag("LoggerInterface")
+class MockLogger implements LoggerInterface {
```

Comment:
Should we have @moduleScoped ? Just to be consistent with what we are saying below ?

Reply:
Since I'm only introducing the concept after, I think to avoid confusion I'll leave it out for this example.